src/Orleans/Async/UnobservedExceptionsHandlerClass.cs

```diff
@@ -26,6 +27,11 @@ static UnobservedExceptionsHandlerClass()
             }
         }
 
+        internal static void InitLogger(ILoggerFactory loggerFactory)
+        {
+            logger = loggerFactory.CreateLogger<UnobservedExceptionsHandlerClass>();
```

Comment:
Now that we have different loggers for each instance of client and silos, this whole static class should go away I believe. And we also talked about not logging unobserved exceptions in the client. Not sure it needs to be addressed in this PR though

Reply:
I actually made it non-static and injected it into DI as singleton. But I reverted that change since I wasn't sure how that would affects perf. We can discuss that in another PR to give it more focus